stream/src/public_key/mod.rs

```diff
@@ -45,26 +45,25 @@
 //! ## Encryption
 //!
 //! During the handshake (described above), a shared x25519 secret is established using a
-//! Diffie-Hellman Key Exchange. This x25519 secret is then used to create a ChaCha20-Poly1305
-//! cipher for encrypting all messages exchanged with the peer.
+//! Diffie-Hellman Key Exchange. This x25519 secret is then used in-conjunction with the handshake
+//! data in a key-derivation-function to create a pair of ChaCha20-Poly1305 ciphers. One cipher per
+//! direction allows for encryption of all messages.
 //!
-//! Each peer maintains a pair of ChaCha20-Poly1305 nonces (12 bytes), one for itself and one for
-//! the other. Each nonce is constructed using a counter that starts at either 1 for the dialer, or
-//! 0 for the listener. For each message sent, the relevant counter is incremented by 2, ensuring that
-//! the two counters have disjoint nonce spaces.
+//! Each direction of communication also uses a 12-byte nonce derived from a counter that is
+//! incremented for each message sent. This provides for a maximum of 2^96 messages per sender,
+//! which is sufficient for all practical use cases. This approach ensures well-behaving peers, as
```

Comment:
nit: prefer the previous language for how many messages this is (rather than just saying "practical use cases")